packages/SwingSet/src/liveslots/virtualObjectManager.js

```diff
@@ -487,11 +539,40 @@ export function makeVirtualObjectManager(
       if (initializing) {
         cache.remember(innerSelf);
       }
-      const representative = Far(tag, actualize(wrappedData));
+      const self = actualize(wrappedData);
+      let representative;
+      let exposedRepresentative;
+      const facetiousness = assessFacetiousness(self);
+      switch (facetiousness) {
+        case 'one': {
+          representative = Far(tag, self);
+          vrm.checkOrAcquireFacetNames(kindID, null);
+          exposedRepresentative = representative;
+          break;
+        }
+        case 'many': {
+          exposedRepresentative = {};
+          representative = [];
+          const facetNames = Object.getOwnPropertyNames(self).sort();
+          vrm.checkOrAcquireFacetNames(kindID, facetNames);
+          for (const facetName of facetNames) {
+            const facet = Far(`${tag} ${facetName}`, self[facetName]);
+            exposedRepresentative[facetName] = facet;
+            representative.push(facet);
+          }
+          harden(exposedRepresentative);
```

Comment:
Before it is `exposedRepresentative`, `assessFacetiousness` cannot be reliable, because unhardened objects are too slippery. Don't try to validate much about unhardened objects --- that's rather hopeless. `assessFacetiousness` need only make a correct decision if `exposedRepresentative` is well behaved. Otherwise, `assessFacetiousness` need only make ***some*** decision. Whatever decision that is, after `exposedRepresentative` is hardened, that decision must either be enforced or the attempt at enforcement must throw. 

For the `'one'` case, the `exposedRepresentative` is `Far`ed, after which you know that it is a hardened passable. For the `'many'` case, if you were hardening in place as mu shim did, then just hardening it doesn't ensure that it is a valid copyRecord, so I suggest adding at least the following check. This will also that it is passable, and that it has no cycles within the pass-by-copy structure.

```suggestion
          harden(exposedRepresentative);
          assertRecord(exposedRepresentative);
```

However, you'd still need to go back over it to ensure that the contents are still the objects that you `Far`ed. Now that I've talked this through, I like what you're doing better than my suggestion. By constructing a new `exposedRepresentative` from scratch yourself starting from a well behaved empty object, by asking `self` each question only once, and by not giving `self` and premature access to `exposedRepresentative`, no further mischief is possible.

I think both the non-defensive `assessFacetiousness` and the defensive code above as written, are fine. Could you didn't take my suggestion:

> On the exposed representative, curious why you don't use self itself as you do for the single facet case above. It doesn't matter so no objection. But if it's otherwise indifferent, it would reduce the cost of making the multi-facetted representative.

Bad suggestion. Bad!

Reply:
Even with your code, adding the `assertRecord(exposedRepresentative);` is belt and suspenders. I leave it to you whether it is worth it.

Reply:
I think you should revisit this in the latest rev.  I did something else before I saw this comment and I think what I did works better than either the way it worked before or what you propose here.

Reply:
BTW, I completely don't follow what you say w.r.t `assessFacetiousness` and lack of hardening.  The job of `assessFacetiousness` is to analyze the object it is given; that analysis remains completely reliable as long as we have not returned control to user code, which is how it is used.  If it determines that the object returned by `actualize` has the right shape, that means it's in a suitable state to extract the parts needed and wrap them with `Far`, which is what the code does.